File: Kleisli/Writer.hs
type Writer a = (a, String)

return :: a -> Writer a
return x = (x, "")

(>=>) :: (a -> Writer b) -> (b -> Writer c) -> (a -> Writer c)
m1 >=> m2 = \a ->
    let (b, s1) = m1 a
        (c, s2) = m2 b
    in (c, s1 ++ s2)
    
notW :: Bool -> Writer Bool
notW b = (b, "not")

is_even :: Int -> Writer Bool
is_even x = (x `mod` 2 == 0, "even")

is_odd :: Int -> Writer Bool
is_odd = is_even >=> notW

main = print (is_odd 7)

File: Kleisli/Writer.py
from collections import namedtuple

Writer = namedtuple('Writer', ['val', 'log'])

def id_writer(x):
    return Writer(x, "")
    
def compose_writer(m1, m2):
    def f(a):
        b, s1 = m1(a)
        c, s2 = m2(b)
        return Writer(c, s1+s2)
    return f
    
def notW(b):
    return (not b, "not")

def is_even(x):
    return (x%2==0, "even")
    
def is_odd(x):
    return compose_writer(is_even, notW)(x)
    
    
print(is_odd(7))
